lib/Interpreter/DeclUnloader.cpp

```diff
@@ -711,10 +736,49 @@ bool DeclUnloader::VisitRedeclarable(clang::Redeclarable<T>* R, DeclContext* DC)
 
   bool DeclUnloader::VisitNamespaceDecl(NamespaceDecl* NSD) {
     // NamespaceDecl: NamedDecl, DeclContext, Redeclarable
-    bool Successful = VisitDeclContext(NSD);
-    Successful &= VisitRedeclarable(NSD, NSD->getDeclContext());
+
+    // When unloading a NamedDecl within an inline namespace within std the
+    // NamedDecl needs to be removed from the parent std as well.
+    // Not sure why, but test/CodeUnloading/NameSpaces.C demonstrates
```

Comment:
The lookup tables of the inline namespaces are in their enclosing parent namespaces. So we should be asking if the NSD is inline and we should get the Enclosing lookup namespace. This should be reproducible outside of the namespace `std` too.